pkg/services/ngalert/api/api_ruler.go

```diff
@@ -452,6 +452,35 @@ func (srv RulerSrv) checkGroupLimits(group apimodels.PostableRuleGroupConfig) er
 //
 //nolint:gocyclo
 func (srv RulerSrv) updateAlertRulesInGroup(c *contextmodel.ReqContext, groupKey ngmodels.AlertRuleGroupKey, rules []*ngmodels.AlertRuleWithOptionals, deletePermanently bool) response.Response {
+	finalChanges, amConfig, err := srv.performUpdateAlertRules(c, groupKey, rules, deletePermanently)
```

Comment:
`updateAlertRulesInGroup` was returning a Response. I extracted the part that did the updates into `performUpdateAlertRules` to reuse.